SLE-793: Dependency simplification

thahnen opened the pull request:
## Summay

In order to fix issues of leaking packages that created conflicts with other Eclipse plug-ins/bundles, the new dependencies on SLCORE are once again shaded/relocated into a single OSGi bundle. The main changes are done on SLCORE; on SLE, there is only relying on the relocated GSON.

The bundle `sonarlint-java-client-osgi` contains the other SLCORE modules as well as the dependencies *LSP4J*, *GSON*, and *ProtoBuf*. This mimics the pre-Sloop logic of having the `sonarlint-core-osgi` bundle.

## Testing

Integration tests running successfully, manually tested different Eclipse installation and plug-in combinations. Testing with COBOL-IDEs will be done on their side, already reached out.

nquinquenel reviewed (approved):
Looks much more cleaner with this single bundle!

thahnen committed:
SLE-793: Dependency simplification

Narrowing down the focus of SLCORE dependencies to only one again while fixing the issue of exposing too much / bundling third-party dependencies on our own Update Site that might be incompatible with some installations.
